src/Egulias/EmailValidator/Parser/Parser.php

```diff
@@ -36,14 +36,18 @@ protected function validateQuotedPair()
     }
 
     /**
-     * @return string the the comment
-     * @throws \InvalidArgumentException
+     * @param int $openedParenthesis
+     * @return string the comment
      */
-    protected function parseComments()
+    protected function parseComments(&$openedParenthesis = 0)
```

Comment:
Instead of passing by reference (which causes hidden side effects), the method can return the number of parenthesis found. What do you think?